cpp/src/arrow/flight/sql/odbc/tests/CMakeLists.txt

```diff
@@ -34,6 +34,7 @@ add_arrow_test(flight_sql_odbc_test
                SOURCES
                odbc_test_suite.cc
                odbc_test_suite.h
+               errors_test.cc
```

Comment:
Are we deliberately wanting connection_test at the bottom?

Reply:
Initially yes as I thought putting `connection_test` at the bottom was a workaround for segfault, but now I think it is not needed. 
I re-ordered order the test files alphabetically in CMakeLists.txt